dns_check/datadog_checks/dns_check/dns_check.py

```diff
@@ -105,23 +110,47 @@ def _check(self, instance):
             self.log.debug('Resolved hostname: {0}'.format(hostname))
             return Status.UP, 'UP'
 
+    def _check_answer(self, answer, resolves_as):
+        ips = [x.strip().lower() for x in resolves_as.split(',')]
+        number_of_results = len(answer.rrset.items)
+
+        assert(len(ips) == number_of_results)
+        if number_of_results > 1:
+            result_ips = []
+            for rip in answer.rrset.items:
+                result = rip.to_text().lower()
+                if result.endswith('.'):
+                    result = result[:-1]
+                result_ips.append(result)
+
+            for ip in ips:
+                assert(ip in result_ips)
+        else:
```

Comment:
not sure the if/else block is worth it, I'd pay the price to build `result_ips` even when there's only one item to get a more readable code in exchange